thirdparty/faiss/faiss/utils/distances.cpp

```diff
@@ -1155,13 +1155,18 @@ void elkan_L2_sse(
         size_t nx,
         size_t ny,
         int64_t* ids,
-        float* val) {
+        float* val,
+        float* data) {
     if (nx == 0 || ny == 0) {
         return;
     }
 
     const size_t bs_y = 1024;
-    float* data = (float*)malloc((bs_y * (bs_y - 1) / 2) * sizeof(float));
+    bool allocate = false;
```

Comment:
Please use `std::unique_ptr<float[]>` + `new` for allocating a local temporary buffer

Reply:
Call resize() to allocate the memory in constructor. Remove the checking here for use_elkan can help ensure the memory is allocated when come here.